client/session/stored_credentials.go

```diff
@@ -23,6 +23,11 @@ const (
 	CredentialsTypeGitHubToken
 )
 
+// String returns the string representation of the type.
+func (t CredentialsType) String() string {
+	return [...]string{"Invalid", "API Key", "GitHub"}[t]
```

Comment:
Today I learned what Array syntax is 😯
```
[...]string{
```
Though, do I understand correctly that []string{...} would be perfectly ok here and this just omits one allocation?

Reply:
Well this is slightly awkward... the truth is that I just copied it from an example 😆 I was trying to search for something like "what's best practice in Go for converting enum values to strings" and found this page: https://yourbasic.org/golang/iota/

Now you've put me on the spot I think you're right. `[...]string{"1", "2"}` allocates an array with a length of two, and `[]string{"1", "2"}` allocates an array with a length of two, and wraps it in a slice from my quick research. The `...` syntax is just used to tell the compiler to automatically set the array length, so `[...]string{"1", "2"}` is equivalent to `[2]string{"1", "2"}`.

My guess is the reasons for doing it like this are:

- To avoid having to update the array length as new items are added.
- To avoid creating a slice when all we're ever doing with the array is directly accessing an element by its index.

I'm guessing that using a slice here would be totally fine as well (in the sense that it's unlikely to really make a different to performance for this application).